src/clj/fluree/db/validation.cljc

```diff
@@ -248,7 +248,8 @@
     ::subject              ::iri
     ::function             [:orn
                             [:string-fn [:and :string [:re #"^\(.+\)$"]]]
-                            [:list-fn [:and list? [:cat :symbol [:* any?]]]]]
+                            [:list-fn [:and list? [:cat :symbol [:* any?]]]]
+                            [:data-fn [:and vector? [:cat [:enum :expr] [:* any?]]]]]
```

Comment:
tiny nitpick, but lists are also data. perhaps this could be called `:vector-fn` or `:expr-pattern`.